### PR TITLE
Introduce a `mutate` method on code-generated entities

### PR DESCRIPTION
**Pull request by jblebrun**

### CODE REVIEW

File: java/arcs/core/entity/EntityBase.kt [hunk 1]
```diff
@@ -30,14 +30,14 @@ open class EntityBase(
     private val schema: Schema
 ) : Entity {
     // Private var _entityId with public getter. Only this class can set this field.
-    private var _entityId: String? = null
+    protected var _entityId: String? = null
```
Comment (csilvestrini) [inline java/arcs/core/entity/EntityBase.kt:33]: Comment doesn't match up.

But also, the reason I did this in the first place was to prevent developers being able to change the entity ID. Is this change necessary?

File: javatests/arcs/core/entity/HandleManagerTestBase.kt [hunk 2]
```diff
@@ -477,6 +477,22 @@ open class HandleManagerTestBase {
         assertThat(updateDeferred.await()).containsExactly(entity1, entity2)
     }
 
+    @Test
+    open fun collection_writeMutatedEntityReplaces() = testRunner {
+        val entity = TestParticle_Entities(text = "Hello")
+        val handle = writeHandleManager
+            .createCollectionHandle(entitySpec = TestParticle_Entities)
```
Comment (csilvestrini) [inline javatests/arcs/core/entity/HandleManagerTestBase.kt:484]: Indenting seems weird. How about breaking after the paren instead?

File: java/arcs/core/entity/EntityBase.kt (hunk 1 above)
Comment (jblebrun) [reply java/arcs/core/entity/EntityBase.kt]: You caught me... I don't like this either, but I was trying to minimize the amount of codegen editing I had to do ;-)

Fixed this up to keep the setters private.

File: src/tools/schema2kotlin.ts [hunk 3]
```diff
@@ -393,9 +405,16 @@ ${classDef}${constructorArguments}${classInterface}
...
+     * Use this method to create a new, distinctly identified copy of the entity. 
+     * Storing the copy will result in a new copy of the data being stored.
+     */`}
     fun copy(${ktUtils.joinWithIndents(this.fieldsForCopyDecl, 14, 2)}) = ${name}(${ktUtils.joinWithIndents(this.fieldsForCopy, 8+name.length, 2)})
-
+    ${this.opts.wasm ? `` : `/** 
+     * Use this method to create a new version of an existing entity.
+     * Storing the mutation will overwrite the existing entity in the set, if it exists.
+     */
+    fun mutate(${ktUtils.joinWithIndents(this.fieldsForCopyDecl, 14, 2)}) = ${name}(${ktUtils.joinWithIndents(fieldsForMutate, 8+name.length, 2)})`}
```
Comment (jblebrun) [inline src/tools/schema2kotlin.ts:417]: This is all a bit bunched up; I was attempting to avoid any changes (including newlines) in the generated wasm files
